src/Uno.UI/UI/Xaml/Style/Generic/Generic.Native.xaml

```diff
@@ -77,6 +69,7 @@
 
 	<xamarin:Style
 		TargetType="Button"
+		xamarin:IsNativeStyle="True"
```

Comment:
Why do native styles need to be flagged ?

Reply:
What do you think about it? The old way of handling native styles had to go because it was 'override them ad hoc in GenericStyles.cs at startup', and GenericStyles.cs is gone.

I opted to explicitly flag native styles in Xaml, like this. They're then registered as a native style by the generated code.

This has two advantages:

1. The `UseUWPDefaultStyles` flag is checked when the Style is first resolved, which cuts down on pseudo-race conditions where the Styles are initialized before the flag is set by user code;
2. Third-party libraries can define native versions of default Styles for custom controls.

I took it one step further and added a `FeatureConfiguration` option for toggling native style usage per control. Curious for feedback on that part specifically well:

```csharp
			/// <summary>
			/// Override the native styles usage per control type.
			/// </summary>
			/// <remarks>
			/// Usage: 'UseUWPDefaultStylesOverride[typeof(Frame)] = false;' will result in the native style always being the default for Frame, irrespective
			/// of the value of <see cref="UseUWPDefaultStyles"/>. This is useful when an app uses the UWP default look for most controls but the native
			/// appearance/comportment for a few particular controls, or vice versa.
			/// </remarks>
			public static IDictionary<Type, bool> UseUWPDefaultStylesOverride { get; } = new Dictionary<Type, bool>();
```

It's kinda useful, but arguably unnecessary since people can also just put `<Style TargetType="Frame" BasedOn="{StaticResource NativeDefaultFrame}"/>` in their Xaml.

Reply:
Ok I see what that means, and yes it makes sense to flag those as such. Thanks for the explanation. Just make sure to insert that in the documentation.

Reply:
Indeed, will do